articles/image-classifier-keras/index.md

```diff
@@ -0,0 +1,484 @@
+
+
+```python
+# Awesome HyperParameters!!!
+TARGET_SIZE = (200, 200)
+BATCH_SIZE = 32
+CLASS_MODE = 'binary'  # for two classes; categorical for over 2 classes
+
+# Connecting the ImageDataGenerator objects to our dataset
+train_generator = dgen_train.flow_from_directory(train_dir,
+                                                 target_size=TARGET_SIZE,
+                                                 subset='training',
+                                                 batch_size=BATCH_SIZE,
+                                                 class_mode=CLASS_MODE)
+
+validation_generator = dgen_train.flow_from_directory(train_dir,
+                                                      target_size=TARGET_SIZE,
+                                                      subset='validation',
+                                                      batch_size=BATCH_SIZE,
+                                                      class_mode=CLASS_MODE)
+test_generator = dgen_test.flow_from_directory(test_dir,
+                                               target_size=TARGET_SIZE,
+                                               batch_size=BATCH_SIZE,
+                                               class_mode=CLASS_MODE)
+```
+
+![alt text](/engineering-education/image-classifier-keras/11.jpg)
+
+The first argument is the path to the dataset. The next parameter is the *target_size*. It will resize all the images to the specified target size of 200x200. The *batch size* defines how many images we want to have in each batch. 
+
+We will use a batch size of 32, and the class mode is either *binary* or *categorical*. *Binary* is for two output classes, while categorical is for more than two classes. Since we are working with only *two* classes, we will set the class mode to *binary*.
+
+The *subset* parameter keeps track of the images we will use for training and validation from the *train_dir*. We don’t need the *subset* parameter for the test generator. We use the subset parameter only if we use validation_split.
+
+**Note:** We only apply image augmentation to the training set. We can normalize the validation and test set as well.
+
+To get the class indices, we use the *class_indices* attribute.
+
+```python
+# Get the class indices
+train_generator.class_indices
+```
+
+![alt text](/engineering-education/image-classifier-keras/12.jpg)
+
+We can also get the shape of our image by using the *image_shape* function.
+
+```python
+# Get the image shape
+train_generator.image_shape
+```
+
+![alt text](/engineering-education/image-classifier-keras/13.jpg)
+
+
+### Build CNN model
+
+```python
+# Building CNN Model
+model = Sequential()
+model.add(Conv2D(32, (5,5), padding='same', activation='relu',
+                input_shape=(200, 200, 3)))
+model.add(MaxPooling2D(pool_size=(2,2)))
+model.add(Dropout(0.5))

```

Comment:
My experience says a dropout of 0.2-0.25 is ideal. Anything above that leads to loss of data and bad training. 0.5 is huge,